src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp

```diff
@@ -390,15 +392,35 @@ void BlockLocalPositionEstimator::update()
 		mavlink_and_console_log_info(&mavlink_log_pub, "[lpe] reinit x");
 	}
 
-	// reinitialize P if necessary
+	// force P symmetry and reinitialize P if necessary
 	bool reinit_P = false;
 
 	for (int i = 0; i < n_x; i++) {
-		for (int j = 0; j < n_x; j++) {
-			if (!PX4_ISFINITE(_P(i, j))) {
+		for (int j = 0; j <= i; j++) {
+			if (_P(i, j) > P_MAX) {
+				// if we aren't measuring terrain state etc.
+				// then it will grow to this max, don't
+				// want to reset here or it will be a
+				// periodic reset
+				_P(i, j) = P_MAX;
```

Comment:
I would still say that it's dangerous to change just a single value without touching the column/row neighbors. We've seen exactly this kind of partial reset cause negative diagonal entries a few update steps later in EKF2. But I agree that it's not desirable to reset the entire covariance matrix just because no terrain measurements are coming in.
Since this is a generic sanity check, I'd vote for zeroing the affected columns and rows.